app/presenters/vendor_api/single_application_presenter.rb

```diff
@@ -78,13 +78,13 @@ def nationalities
       ].map { |n| NATIONALITIES_BY_NAME[n] }.compact
     end
 
-    def course
+    def course_info_for(application_choice_or_course_option)
```

Comment:
What would you think of naming this parameter `course_option`? It's a duck. If we don't like that (i.e. depending on two types having this interface) we could read `course_option` off `application_choice` before passing it in?

Reply:
Done :+1:, it also uncovered N+1 queries that were being not picked up before